src/company_gui.cpp

```diff
@@ -2126,6 +2146,22 @@ struct CompanyInfrastructureWindow : Window
 				break;
 		}
 	}
+	void OnClick([[maybe_unused]] Point pt, WidgetID widget, [[maybe_unused]] int click_count) override
+	{
+		switch (widget) {
+			case WID_CI_HIGHLIGHT_INFRASTRUCTURE:
+				CompanyID window_company_id = (CompanyID)this->window_number;
+				if(_viewport_company_to_highlight_infrastructure != window_company_id) {
```

Comment:
```suggestion
				if (_viewport_company_to_highlight_infrastructure != window_company_id) {
```

Also, some comments noting which branch of the if/else is `/* Highlight tiles of this company */` and `/* Clear tile highlights */` would be nice, for future readability. 🙂